externals/pfm/dynamics/dynamicForms.js

```diff
@@ -0,0 +1,57 @@
+
+/**
+ * Improves forms usability
+ */
+export class DynamicForms {
+
+    /**
+     * Generates content of an html input depending on an html select selected option  
+     * @param {string} field1 id of the source select
+     * @param {string} field2 id of the dependant input
+     * @param {string} apiRoute
+     * name of route to call the backend function returning data to fill dependant select options
+     *  
+     */
+    dynamicFields(field1, field2, apiRoute) {
+        const firstField = document.getElementById(field1);
+        firstField.addEventListener("change", (event) => {
+            let id = event.target.value;
+            const secondField = document.getElementById(field2);
+            const spaceId = document.getElementById("id_space").value;
```

Comment:
you suppose there is a id_space element, should not "expect" things, take it as parameter